### PR TITLE
fondo, notes-up: init

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
@davidak wanted notes-up.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by davidak — APPROVED** [review]
Looks good.

Thank you very much!

**Comment by dtzWill** [comment]
FWIW when trying to run this, it seems to bail (I'm not using full pantheon bits):
```
$ com.github.philip-scott.notes-up                                                                     12:12:34 on 19-02-22
[INFO 12:12:36.547813] Application.vala:155: Notes-Up version: 1.3
[INFO 12:12:36.547849] Application.vala:157: Kernel version: 4.20.11
[FATAL 12:12:36.692385] file /build/source/src/Widgets/Headerbar.vala: line 126: uncaught error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.elementary.Contractor was not provided by any .service files (g-dbus-error-quark, 2)
[FATAL 12:12:36.723962] [Gtk] gtk_widget_get_parent: assertion 'GTK_IS_WIDGET (widget)' failed
[FATAL 12:12:36.724005] [Gtk] gtk_widget_freeze_child_notify: assertion 'GTK_IS_WIDGET (widget)' failed
zsh: segmentation fault (core dumped)  com.github.philip-scott.notes-up
```

Is this something to be resolved at the packaging level, or more intrinsic fault regarding compatibility in other environments?

**Comment by worldofpeace** [comment]
I'm pretty sure `fondo` would only ever be useful in pantheon because of their greeter setup, etc.

But I think for `notes-up` that there's a special build flag that disables contractor support, because outside pantheon you won't have that interface.

So i probably should add a subpackage maybe that has `-Dnoele=1` in `cmakeFlags`.

**Comment by dtzWill** [comment]
> So i probably should add a subpackage maybe that has `-Dnoele=1` in `cmakeFlags`.

As you see fit, and thank you.  I mostly wanted to be sure it didn't indicate a dep that needed to be propagated or otherwise was a packaging bug.

Sounds like it's good as -is :).

**Comment by dtzWill** [comment]
Sorry for hijacking this a bit, but just FYI for anyone with this sort of issue:

Simply adding `pantheon.contractor.enable = true;` seems to have done the trick, no more crashing and doesn't seem to be intrusive or require full buy-in.  But I wouldn't expect too much re:functionality since it likely is meant for other components too.  Anyway, thanks!

**Comment by worldofpeace** [comment]
> Sorry for hijacking this a bit, but just FYI for anyone with this sort of issue:
> 
> Simply adding `pantheon.contractor.enable = true;` seems to have done the trick, no more crashing and doesn't seem to be intrusive or require full buy-in. But I wouldn't expect too much re:functionality since it likely is meant for other components too. Anyway, thanks!

No hijack done :smile: this should work wherever by default https://github.com/NixOS/nixpkgs/pull/56172/commits/366945ac37981b2aa9a4765544a79b99a5d6869c

**Comment by dtzWill** [comment]
:heart: thanks! Yes this looks great!

File: pkgs/applications/office/notes-up/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    sqlite
+    webkitgtk
+  ];
+
+  # Whether to build with contractor support (Pantheon specific)
+  cmakeFlags = if withPantheon then null else [ "-Dnoele=yes" ];
+
+  meta = with stdenv.lib; {
+    description = "Markdown notes editor and manager designed for elementary OS"
+    + stdenv.lib.optionalString withPantheon "- built with Contractor support";
```
Comment (worldofpeace) [inline pkgs/applications/office/notes-up/default.nix]: Not sure if this will work.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg eval

**Comment by worldofpeace** [comment]
ping @jtojnar

File: pkgs/applications/office/notes-up/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/applications/office/notes-up/default.nix]: I do not see why it should not. But maybe add a space:

```suggestion
    + stdenv.lib.optionalString withPantheon " – built with Contractor support";
```

**Comment by jtojnar** [comment]
Looks about right.

File: pkgs/applications/office/notes-up/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/applications/office/notes-up/default.nix]: Right, concatenating strings 101 :smile:

**Comment by worldofpeace** [comment]
Thanks for the reviews everyone :+1:

**Pull request merged**